app/javascript/packages/rails-i18n-webpack-plugin/rails-i18n-webpack-plugin.js

```diff
@@ -162,16 +92,28 @@ class RailsI18nWebpackPlugin extends ExtractKeysWebpackPlugin {
    * @param {string} locale
    * @param {MissingStringCallback} onMissingString
    *
-   * @return {Promise<string>}
+   * @return {Promise<string|Record<string, string>>}
    */
   async resolveTranslation(key, locale, onMissingString = this.options.onMissingString) {
-    const keyPath = getKeyPath(key);
-    const domain = getKeyDomain(keyPath);
-    const localeData = await this.getLocaleData(domain, locale);
+    const localeData = await this.getLocaleData(locale);
+
+    /** @type {undefined | string | Record<string,string>} */
+    let translation = localeData?.[key];
+
+    // Prefix search localeData, used in ".one", ".other" keys
+    if (translation === undefined && typeof localeData === 'object') {
+      const prefix = `${key}.`;
+      const prefixedEntries = Object.entries(localeData)
+        .filter(([localeDataKey]) => localeDataKey.startsWith(prefix))
+        .map(([localeDataKey, value]) => [localeDataKey.replace(prefix, ''), value]);
```

Comment:
Started thinking through an edge-case like `prefix = 'foo.bar.'` replacing in a key of `'foo.bar.foo.bar'`, but I guess we can rely on `replace` to be lazy and only replace the first match.

Otherwise we could do a RegExp match/replace on `^`, or return substring of `localeDataKey.slice(prefix.length)`.

Reply:
Will leave as-is for now and rely on `replace` being lazy for the time being